server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java

```diff
@@ -36,13 +37,12 @@ public class JavaDateMathParser implements DateMathParser {
 
     private final JavaDateFormatter formatter;
     private final String format;
-    private final JavaDateFormatter roundupParser;
+    private final Function<String, TemporalAccessor> roundupParser;
 
-    JavaDateMathParser(String format, JavaDateFormatter formatter, JavaDateFormatter roundupParser) {
+    JavaDateMathParser(String format, JavaDateFormatter formatter, Function<String, TemporalAccessor> roundupParser) {
```

Comment:
woudl it be more obvious if we replace `JavaDateFormatter formatter` with ` Function<String,TemporalAccessor> parser` too? 
creating a `JavaDateMathParser` is from `toDateMathParser()` so it would be an impl detail anyway.

Reply:
Good idea